client/shared/src/mem.rs

```diff
@@ -12,18 +12,56 @@ const fn MiB(n: usize) -> usize {
 
 const STACK_ALIGNMENT: usize = 1024 * 63;
 
+/// The size of the stack. Its start is at address 0
 pub const GRAPHIC_STACK_SIZE: usize = MiB(4) + STACK_ALIGNMENT;
+
+/// The address of the Allocator structures (size: 1MiB)
 pub const ALLOCATOR_AREA_START: usize = GRAPHIC_STACK_SIZE;
-pub const SHARED_BUFFER_AREA_START: usize = ALLOCATOR_AREA_START + MiB(1);
+
+/// The address memory communication area. (size: 1MiB)
+/// It contains data needed for communication between main thread and logic thread.
+pub const COMMUNICATION_MEMORY_START: usize = ALLOCATOR_AREA_START + MiB(1);
+
+/// The address of the double buffer (size: target dependent)
+pub const SHARED_BUFFER_AREA_START: usize =
+    COMMUNICATION_MEMORY_START + core::mem::size_of::<CommunicationMemory>();
+
+/// The logic heap address (size: 32MiB)
 pub const LOGIC_ALLOCATION_AREA_START: usize =
-    SHARED_BUFFER_AREA_START + std::mem::size_of::<Buffer>();
+    SHARED_BUFFER_AREA_START + core::mem::size_of::<Buffer>();
+
+/// The graphics heap address (size: 32MiB)
```

Comment:
Do we need a graphics heap, could it be smaller?

Reply:
I'm not sure if I can completely abstain from the graphics heap, but I agree that 32MiB is too big. I propose, we should track a list of all allocations in the complete frontend code (also the game-engine). That would make memory size estimation a lot easier.

Reply:
agreed